### PR TITLE
Fixed UI Inconsistencies with Figma (LocationCards and ApartmentCard)

### PR DESCRIPTION
**Pull request by kea-roy**
### Background

After feedback from designers and devs, our team found many inconsistencies between the designs on Figma and the implemented design in the main branch. This pull request fixes inconsistencies found on the LocationCards and ApartmentCard components.

### Changes

This pull request contains the following changes:
- [X] Fixed LocationCard and LocationCards styling to match the Figma design. 
     - [X] Adjusted the spacing
     - [X] Adjusted the drop shadow
     - [X] Added the hover effect
     - [X] Adjusted the font styling and margin/padding values for responsiveness.
     - [X] Adjusted the image styling to account for rounded SVGs, so the style matches the Figma on all screen sizes.
- [X] Fixed ApartmentCard styling to match the Figma design. 
     - [X] Adjusting the grid layout
     - [X] Adjusting the spacing
     - [X] Adjusting the font styling
     - [X] Adjusting the responsiveness
     - [X] Reworking the HeartRating component to support different numerical font sizes
     - [X] Added hover effect for saved icon
- [X] Added documentation for LocationCard, LocationCards, and HeartRating components.

## Test Plan 

Compared design to Figma using Chrome's developer tools. I tested out the design on desktop-sized screens and mobile screens. Below are some images comparing the design in Figma to the coded design.

### LocationCards Old Design
00 HomePage locationcards old design
### Location Cards Figma
01 HomePage locationcards figma
### Location Cards New Design
02 Homepage locationcards new code

### ApartmentCard Old Design
00 HomePage ApartmentCard old design
### ApartmentCard Figma
01 HomePage ApartmentCard figma
### ApartmentCard New Design
02 Homepage ApartmentCard new code

### ApartmentCard Mobile Old Design
00 HomePage ApartmentCard Mobile old design
### ApartmentCard Mobile Figma
01 HomePage ApartmentCard Mobile figma
### ApartmentCard Mobile New Design
02 Homepage ApartmentCard Mobile new code

### CODE REVIEW

File: frontend/src/components/Home/LocationCard.tsx [hunk 1]
```diff
@@ -11,20 +11,38 @@ const LocationCard = ({ photo, location }: Props): ReactElement => {
...
+    },
+    imgCropper: {
+      clipPath: 'inset(1%)',
+      scale: 1.03,
+    },
+    imgContainer: {
+      overflow: 'hidden',
+    },
+    textBox: {
+      height: '61.75px',
```
Comment (ggsawatyanon) [inline frontend/src/components/Home/LocationCard.tsx:25]: It's great to see your attention to detail to match the figma but in the future, try to use whole numbers for styling (e.g. 61 px) or maximum 1 decimal place (e.g. 61.8 px) in exceptional cases where finetuning is really needed. This ensures cleaner code and more predictable design.

**Review by ggsawatyanon — DISMISSED** [review]
You did a good job matching the UI inconsistencies with the Figma but I noticed that the height of the white part of the location cards is too long. Could you fix this? It is especially noticable on mobile version.
Figma:
image
On mobile:
image
On desktop:
image

**Comment by kea-roy** [comment]
> You did a good job matching the UI inconsistencies with the Figma but I noticed that the height of the white part of the location cards is too long. Could you fix this? It is especially noticable on mobile version. Figma: image On mobile: image On desktop: image

Thanks Grace for pointing out the issue, I resolved the issue and updated the pull request!

**Comment by kea-roy** [comment]
## Commit Changes
1. Added hover effect for saved button on ApartmentCard

**Comment by kea-roy** [comment]
## Commit Changes

1. Simplified CSS code to resolve merge conflicts by removing unused CSS classes.

**Comment by kea-roy** [comment]
## Commit Changes
1. Fixed Save Ribbon Styling
2. Simplfied Save Ribbon ternary operators for readability
3. Added save ribbon hover animation

**Review by ggsawatyanon — APPROVED** [review]
Good job on fixing the inconsistencies and the ribbon icon. In the future, try to avoid reverting a merge commit because it can cause a lot of issues when trying to merge main into the branch. You can reset the merge commit before pushing to remote.

**Pull request merged**